### PR TITLE
[WIP] Automate add new mentors

### PR DESCRIPTION
**Pull request by nverbic**
## Description
 
Feature is not completed - Work in progress.

1. Implement new functionality - update the existing mentors.yml with newly added mentors from the xlsx file:
     def get_new_mentors_data_in_yml_format(yml_file_path, xlsx_file_path):
       - compare names of the mentors from mentors.yml file with the names of the mentors in the xlsx file
       - check for new mentors added after the April 2024 (in code, using the number of the table row of the mentor added on 
         03.04.2024. )
       -  use indexing from the mentors.yml - continue the indexing from the latest (highest) index
       - TODO: might be better to use date and not the row number (in case if older mentors can be removed from the xlsx file)
       - TODO: Special case: mentors with the same name
    
    def read_yml_file(file_path):
        - read yml file and return dictionary

2. Update existing functionality - create new menotrs.yml
     def get_all_mentors_data_in_yml_format(xlsx_file_path):
        - creates new mentors.yml file with all mentors from the xlsx file
        - TODO: read mentor index from the current mentors.yml file and use it when creating new file

     def write_yml_file(file_path, mentors_data, mode):
       - mode: append/write option
       - yml indenting rules are used when creating new file
   
     def xlsx_to_yaml_parser(mentor_row, mentor_index):
        - add availability: []  to the mentor's data
      
     def extract_numbers_from_string(text_arg, get_max_value=True):
        - if the text_arg in NAN, return 0 and not an empty string (probelm noticed when testing - mentors.html page was not 
           showing)
     

Implements required feature - append new mentors from the Excel table to the existing mentors.yml file.

## Change Type
- [ ] Bug Fix
- [x] New Feature
- [ ] Code Refactor
- [ ] Mentor Update
- [ ] Documentation
- [ ] Other

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] I checked and followed the [contributor guide](CONTRIBUTING.md) 
- [x] I have tested my changes locally.
- [ ] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .idea
 _site
 *.DS_Store
-node_modules/*
\ No newline at end of file
+node_modules/*
```
Comment (dricazenck) [inline .gitignore:4]: it has conflicts. I think you have to update your branch

File: tools/automation.py [hunk 2]
```diff
@@ -109,94 +111,169 @@ def get_mentorship_type(mentorship_type_str):
     return ""
 
 
-def write_mentors_yml_file(mentors_data):
+def write_yml_file(file_path, mentors_data, mode):
     """
-    Create mentors.yml file
+    Create new or append to mentors.yml file
     :mentors_data: list of dictionaries
+    :mode: 'w' or 'a'
```
Comment (dricazenck) [inline tools/automation.py:118]: would be nice to say what w means and a mean, by using constants ;)

Comment (dricazenck) [reply tools/automation.py:118]: you could use something like this: 

```
from enum import Enum

# Define an enumeration class
class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
```

File: tools/automation.py [hunk 3]
```diff
@@ -68,17 +68,19 @@ def extract_numbers_from_string(text_arg, get_max_value=True):
...
-        digits = [int(num) for num in re.findall(r"\d+", text_arg)]
-        if digits:
-            if get_max_value:
-                return max(digits)
-            return digits
-
-    return ""
+    # TODO: If the field hours in xlsx is NAN, what should be the default: 0 or 1
+    if not pd.isna(text_arg):
+        if isinstance(text_arg, str):
```
Comment (dricazenck) [inline tools/automation.py:73]:
```suggestion
        if not pd.isna(text_arg) and isinstance(text_arg, str):
```

File: .gitignore (hunk 1 above)
Comment (nverbic) [reply .gitignore]: Reverted

File: tools/automation.py (hunk 2 above)
Comment (nverbic) [reply tools/automation.py:118]: Added! Thank you for the idea.

File: tools/automation.py (hunk 3 above)
Comment (nverbic) [reply tools/automation.py]: Updated

**Pull request closed without merging**